libs/clj-icat-direct/src/clj_icat_direct/icat.clj

```diff
@@ -38,6 +39,19 @@
   [query & args]
   (k/exec-raw icat [query args] :results))
 
+(defn- run-transaction
+  "Runs the set of passed-in query strings+args in a transaction, returning
+  what the last one returns. This is intended to be used to create temporary
+  tables to circumvent the fact postgresql is bad at doing estimates on CTEs."
+  [& queries]
+  (db/transaction
+    (let [side-effects (drop-last 1 queries)
+          final-query-and-args (last queries)]
+      (doseq [q side-effects]
+        ; uses jdbc/execute! because this can run DDL statements like CREATE TEMPORARY TABLE
```

Comment:
I used `run-query-string` (and therefore `k/exec-raw`) initially, but got an error "PSQLException No results were returned by the query. org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery (AbstractJdbc2Statement.java:303)" (I did a search for that, which is where I pulled it from, so might be reformatted). The searching I did suggested that the issue was using the wrong jdbc function. In particular, `exec-raw` calls out to `korma.db/do-query` which calls out to `korma.db/exec-sql`, which uses `query`. The stuff I saw recommended using `executeUpdate`, where `query` uses `executeQuery` (IIRC); nothing in korma uses `executeUpdate`. But now that I'm looking in the morning, it looks like maybe there's something about configuring whether results are returned, which uses something that might also work. I'll test if turning off results fixes this.